src/redux/slices/onboarding.ts

```diff
@@ -0,0 +1,84 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from '@/redux/RootState';
+import SliceName from '@/redux/types/SliceName';
+import OnboardingGroup from '@/types/OnboardingGroup';
+
+type ActiveStepPayload = { group: OnboardingGroup; index: number };
+
+export type OnboardingState = {
+  isChecklistVisible: boolean;
+  checklistDismissals: number;
+  lastChecklistDismissal: Date | null;
+  activeStep: ActiveStepPayload;
+  groups: Partial<
+    Record<
+      OnboardingGroup,
+      {
+        completedSteps: number[];
+      }
+    >
+  >;
+};
+
+const initialState: OnboardingState = {
+  isChecklistVisible: true,
+  checklistDismissals: 0,
+  lastChecklistDismissal: null,
+  activeStep: {
+    group: OnboardingGroup.HOMEPAGE,
+    index: 0,
+  },
+  groups: {},
```

Comment:
nit: should not we call this `completedGroups` since we only store info related to completed groups?